### ISSUE TITLE
[java] UseTryWithResources can flag close calls prior to catch block as well.

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2158]
**Affects PMD Version:** 
6.20.0
**Rule:**
UseTryWithResources

**Description:**
This rule can have an additional optional property to catch close statements within the try block as well. Or not have one but catch it anyway.

**Code Sample demonstrating the issue:**
```java
public class TryWithResources {
    public void run() {
        
        try {
          InputStream  in = openInputStream();
            int i = in.read();
in.close();
        } catch (IOException e) {
            e.printStackTrace();
        } 
}
}
```

**Running PMD through CLI/Ant.**

**Comment by jsotuyod** [comment]
This sounds like a completely different rule…

The `UseTryWithResources` is a best practice rule, to use modern syntax to reduce boilerplate.

What you describe is actually a bug. Closing a resource outside the finally is potentially leaking if any exceptions occurs between open and close. Sounds like something that [`CloseResource`](https://pmd.github.io/pmd-6.24.0/pmd_rules_java_errorprone.html#closeresource) should handle.

I'm pretty sure that rule actively looks for `finally` blocks, but this may be a false negative… Are we sure `CloseResource` is not flagging this?

**Comment by Fernal73** [comment]
You're right , @jsotuyod , this is covered by AutoClosable in the later default configuration. For some reason, I was sticking to the old behaviour.

Closing this issue.....
